pkg/upload/android-ndk.go

```diff
@@ -24,9 +25,18 @@ type AndroidNdkMapping struct {
 func ProcessAndroidNDK(apiKey string, applicationId string, androidNdkRoot string, appManifestPath string, paths []string, projectRoot string, variant string, versionCode string, versionName string, endpoint string, failOnUploadError bool, retries int, timeout int, overwrite bool, dryRun bool) error {
 
 	var fileList []string
+	var processedFileList []string
 	var mergeNativeLibPath string
 	var err error
 
+	tempDir, err := os.MkdirTemp("", "bugsnag-cli-ndk-*")
```

Comment:
I'd prefer to see this done closer to the code that uses it – a lot can happen before the tmp dir is needed and potentially it isn't needed at all if .sym files have been provided.

Suggest adding creation of the directory to the file-loop - if the variable is nil when you need to do an objcopy, then create it first:

```
tempDir = nil
		for _, file := range fileList {
			if strings.HasSuffix(file, ".so.sym") {
				processedFileList = append(processedFileList, file)
			} else if filepath.Ext(file) == ".so" {

				log.Info("Extracting debug info from " + filepath.Base(file) + " using objcopy")

// if tempDir == nil
//   tempDir = MkdirTemp("", "bugsnag-cli-ndk-*")

				outputFile, err := android.Objcopy(objCopyPath, file, tempDir)
...
```

Nit - I'd also not call it tempDir, but workingDir or similar - it describes it's purpose rather than what it is (perhaps it might change in future to a user configurable dir).